### PR TITLE
add check for available service provider - `azurerm_bot_connection`

### LINKED ISSUE #6873: azurerm_bot_connection Azure Active Directory
### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.5
+ provider.azuread v0.8.0
+ provider.azurerm v2.3.0

Your version of Terraform is out of date! The latest version
is 0.12.24. You can update by downloading from www.terraform.io/downloads.html
```

### Expected Behavior

Creation of the **BOT** with the **Teams** Channel and the **OAuth** connection set to `Azure Active Directory` as service provider

### Actual Behavior

The Chatbot and the Channel registration work, but when I try to add the Connection setting the Service Provider Name to : `Azure Active Directory` I get an error :

botservice.BotConnectionClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidBotData" Message="Invalid Service Provider Name. "

The name of the provider should be fine because I got it from the export of the ARM template that I previously manually created, It looks like an API problem but I'm not sure.

### Steps to Reproduce

```
resource "azurerm_bot_channels_registration" "bot_channel_dev" {
  name                                  = "test-bot"
  location                              = "global"
  resource_group_name                   = "RESOURCE_GROUP"
  sku                                   = "S1"
  microsoft_app_id                      = "APP_ID"
  developer_app_insights_key            = "APP_INSIGHT_KEY"
  developer_app_insights_application_id = "APP_INSIGHT_APP_ID"
}

resource "azurerm_bot_channel_ms_teams" "bot_channel_ms_teams_dev" {
  bot_name            = "${azurerm_bot_channels_registration.bot_channel_dev.name}"
  location            = "${azurerm_bot_channels_registration.bot_channel_dev.location}"
  resource_group_name = "${azurerm_resource_group.resource_group_dev.name}"
}

resource "azurerm_bot_connection" "bot_channel_ad_connection_dev" {
  name                  = "testBotConn"
  bot_name              = "${azurerm_bot_channels_registration.bot_channel_dev.name}"
  location              = "${azurerm_bot_channels_registration.bot_channel_dev.location}"
  resource_group_name   = "${azurerm_resource_group.resource_group_dev.name}"
  service_provider_name = "Azure Active Directory"
  client_id             = "${var.dev_1_bot_app_id}"
  client_secret         = "${var.dev_1_bot_app_secret}"
  scopes                = "openid OnlineMeetings.ReadWrite"
  
  parameters = {
    loginUri    = "https://login.microsoftonline.com"
    grantType   = "authorization_code"
    tenantId    = "${var.tenant_id}"
    resourceUri = "https://graph.microsoft.com/"
  }
}

```
### References

https://www.terraform.io/docs/providers/azurerm/r/bot_connection.html

### PR DESCRIPTION
**Pull request by njuCZ**
fix #6873

the error message from the service is not user-friendly.

add check for available service provider

### CODE REVIEW

File: azurerm/internal/services/bot/bot_connection_resource.go [hunk 1]
```diff
@@ -112,9 +112,37 @@ func resourceArmBotConnectionCreate(d *schema.ResourceData, meta interface{}) er
...
+		name := provider.Properties.ServiceProviderName
+		if serviceProviderName == *name {
+			serviceProviderFound = true
+			break
+		}
+		availableProviders = append(availableProviders, *name)
+	}
+
+	if !serviceProviderFound {
+		return fmt.Errorf("invalid Service Provider Name. available service providers are: %+v", availableProviders)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/bot/bot_connection_resource.go:140]: this is likely more readable as a CSV:

```suggestion
		return fmt.Errorf("the Service Provider %q was not found. The available service providers are %s", strings.Join(availableProviders, ","))
```

File: azurerm/internal/services/bot/bot_connection_resource.go [hunk 2]
```diff
@@ -112,9 +112,37 @@ func resourceArmBotConnectionCreate(d *schema.ResourceData, meta interface{}) er
...
+	serviceProviderFound := false
+	var availableProviders []string
+
+	serviceProviders, err := client.ListServiceProviders(ctx)
+	if err != nil {
+		return fmt.Errorf("listing Bot Connection service provider: %+v", err)
+	}
+
+	if serviceProviders.Value == nil {
+		return fmt.Errorf("no available service provider %+v", serviceProviders)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/bot/bot_connection_resource.go:125]: this isn't an actionable error for users, so I guess we can make this:

```suggestion
		return fmt.Errorf("no service providers were returned from the Azure API")
```

**Comment by njuCZ** [comment]
@tombuildsstuff thanks for your suggestion, I have modified the PR. Please have a look again

**Comment by tombuildsstuff** [comment]
@njuCZ running the test for this, this fails:

```
=== RUN   TestAccAzureRMBotChannelsRegistration
=== RUN   TestAccAzureRMBotChannelsRegistration/connection
=== RUN   TestAccAzureRMBotChannelsRegistration/connection/basic
=== RUN   TestAccAzureRMBotChannelsRegistration/connection/complete
TestAccAzureRMBotChannelsRegistration/connection/complete: testing.go:640: Step 0 error: errors during apply:
Error: the Service Provider "Salesforce" was not found. The available service providers are wunderlist,google,pinterest,appFigures,facebook,SkypeForBusiness,outlook,SharePointOnline,Aadv2,linkedin,trello,SharepointServer,oauth2,slack,zendesk,DynamicsCrmOnline,Aad,smartsheet,flickr,Office365,onedrive,basecamp,instagram,mailchimp,Office365User,echosign,live,oauth2generic,spotify,tumblr,AWeber,marketo,dropbox,box,yammer,intuit,uservoice,salesforce,todoist,github,docusign,stripe,bitly,lithium,sugarcrm
```

can we update this to be a case-insensitive comparison?

**Comment by njuCZ** [comment]
Hi @tombuildsstuff sorry for the acctest. I made a change to send `ServiceProviderID` rather than `ServiceProviderDisplayName`, because I thought it might be more stable. Please have a look again

image

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-06-11 at 18 20 53

**Pull request merged**